src/irmin-pack/pack_value.ml

```diff
@@ -0,0 +1,59 @@
+open! Import
+include Pack_value_intf
+
+module Kind = struct
+  type t = Commit | Contents | Inode | Node
+
+  let to_magic = function
+    | Commit -> 'C'
+    | Contents -> 'B'
+    | Inode -> 'I'
+    | Node -> 'N'
+
+  let of_magic = function
+    | 'C' -> Commit
+    | 'B' -> Contents
+    | 'I' -> Inode
+    | 'N' -> Node
+    | c -> Fmt.failwith "Kind.of_magic: unexpected magic char %C" c
```

Comment:
Regarding the approach in #1440 : 
https://github.com/mirage/irmin/blob/dc9eda665b641d54f50fb656e203bceaf7bcebcf/src/irmin-pack/inode.ml#L43-L54

This means that we'll need to edit `Kind.t` to be 

```ocaml
type version  = V0 | V1
type t = { t : t; version : version }

let magic t =
  match t.version with
  | V0 -> ( match t.t with Inode -> 'I' | ...)
  | V1 -> ( match t.t with Inode -> 'J' | ...)
```

something like this, right?

Reply:
Yes, storing the version inside `Pack_value` might be a good idea

Reply:
Agreed. I think it makes sense to inline the `version` inside the `Node` and `Inode` cases specifically – as those are the values being versioned – but we can discuss that when the time comes.